pkg/controllers/state/suite_test.go

```diff
@@ -159,28 +159,30 @@ var _ = Describe("Pod Ack", func() {
 		setTime := cluster.PodSchedulingSuccessTime(nn)
 		Expect(setTime.IsZero()).To(BeTrue())
 
-		cluster.MarkPodSchedulingDecisions(ctx, nil, map[string][]*corev1.Pod{"n1": {pod}})
+		cluster.MarkPodSchedulingDecisions(ctx, nil, map[string][]*corev1.Pod{"n1": {pod}}, map[string][]*corev1.Pod{"nc1": {pod}})
 		setTime = cluster.PodSchedulingSuccessTime(nn)
 		Expect(setTime.IsZero()).To(BeFalse())
 
 		newTime := cluster.PodSchedulingSuccessTime(nn)
 		Expect(newTime.Compare(setTime)).To(Equal(0))
+		Expect(cluster.PodNodeClaimMapping(nn)).To(BeEquivalentTo("nc1"))
 	})
-	It("should delete pod schedulable time if we get error for the pod", func() {
+	It("should delete pod schedulable time and pod to nodeClaim mapping if we get error for the pod", func() {
```

Comment:
You need another test here that deletes the pod when we are clearing the mappings -- frankly, can we double-check that we are clearing all of these mappings generally? Tracking all of this info is tricky business and we need to make sure that we are properly cleaning things up when they no longer need to be tracked